### PR TITLE
chore(clayui.com): update clay static dependencies

### LINKED ISSUE #2891: Size of dependency graph
Just planting this one as a seed for discussion and thought, although it may not lead to any action. The question:

> **What can we (or should we) do about the size of the Clay dependency graph?**

---

The Clay dependency graph is very large.

The `yarn.lock` has 23k lines.

`yarn audit` reports:

```
13427 vulnerabilities found - Packages audited: 960275
Severity: 11 Low | 11 Moderate | 13405 High
```
(Note that it is obviously some double-counting there due to some kind of Yarn bug; there are 80k files in the `node_modules` directory and it must be counting the same packages repeatedly.)

Running `yarn outdated` 108 outdated dependencies:

```
Package                          Current      Wanted  Latest     Workspace                Package Type           URL                                                                                          
@babel/cli                       7.4.4        7.8.4   7.8.4                               devDependencies        https://babeljs.io/                                                                          
@babel/core                      7.4.5        7.8.4   7.8.4                               devDependencies        https://babeljs.io/                                                                          
... 104 lines omitted ...
yeoman-generator                 0.19.2       0.19.2  4.5.0      generator-clay-component dependencies           http://yeoman.io                                                                             
yosay                            1.2.1        1.2.1   2.0.2      generator-clay-component dependencies           http://yeoman.io 
```

[Full output in this gist](https://gist.github.com/wincent/356415c845f4502fb0a39d101564b5b8)

Screenshot showing the output in all of its glorious red/yellow:

clay___yarn

Obviously, all of these dependencies bring some value, otherwise we wouldn't have added them. Most — but not all — of them are `devDependencies`. But I wanted to bring up the size of the graph because these dependencies are not free:

- The larger the graph, the higher the cost of keeping it up to date (something sadly necessitated by security problems, which mean that we don't upgrade because we need access to new features or bug fixes, but because we're forced to by an audit).
- Of the 108 outdated dependencies, 50 of them are semver-breaking major updates. The cost of updating these is hard to evaluate but probably pretty large. At some point, the cost may start to exceed the associated benefits.

Just to put things in perspective, liferay-portal itself "only" has an 18k-line lockfile (about 82% of the size of Clay's).

As I said above, I don't know if any specific action needs to or will be taken as the result of this ticket, but I just wanted to raise awareness about the cost-benefit trade-off involved in maintaining this dependency graph. We might want to check to see if there are any non-essential dependencies, or dependencies that aren't pulling their weight. Or we might merely keep this discussion in mind the next time a PR proposes to add another dependency to the graph.

Sighed, your grumpy, dependency-allergic curmudgeon, @wincent.

clint-eastwood

### PR DESCRIPTION
**Pull request by bryceosterhaus**
fixes #2891

@pat270 here are my updates so that we can better utilize clayui.com for dev.

Updates include..

- Should be able to make changes to clay-css source and see live updates on clayui.com dev site. Run dev site by `cd clayui.com && yarn develop`
    - Note that the static images and icons still are required to be copied over at build time
- Added an atlas/base toggle
Screen Shot 2020-06-17 at 3 32 14 PM

- Added persistence for the css tabs between refresh. Just need the #css hash appended, this will happen when you click the css tab.
Screen Shot 2020-06-17 at 3 32 23 PM

There is still a handful of cleanup we could do, but wanted to try this for now. Let me know what you think

### CODE REVIEW

**Comment by diegonvs** [comment]
About 8f7d489, I loved it! :)

**Comment by matuzalemsteles** [comment]
This is very cool, we can make it visible only in a development environment, to ensure that these things do not go into production.

I had an idea about this at the beginning of making clayui.com for Patrick to use, but without success 🥺, but I think it's still possible to use the base I had before, the idea is to always try to check the `GATSBY_CLAY_NIGHTLY` that is used in development environment still as `false` but that Patrick or anyone else could activate to `true` to do CSS tests, changing this in `.env.development`, the idea was also for the creation of smoke test pages that Patrick used it on his website, just activating `nightly` in the frontmatter of the `.md`. In other words, everything that is under verification of `GATSBY_CLAY_NIGHTLY` will only be shown in this environment without impacting local production or development for example.

**Comment by jbalsas** [comment]
> This is very cool, we can make it visible only in a development environment, to ensure that these things do not go into production.

I like what I'm hearing!

tellmemore

**Comment by bryceosterhaus** [comment]
@matuzalemsteles @jbalsas updated and added the dev check for rendering the settings toggle. Once this is merged, we can go through and add any pages that might be missing from the test site.

**Comment by bryceosterhaus** [comment]
@pat270 let me know if you want anything else added for this

File: clayui.com/gatsby-browser.js [hunk 1]
```diff
@@ -3,7 +3,20 @@
...
+
+if (!devEnv || showAtlas === 'true' || showAtlas === null) {
+	require('@clayui/css/src/scss/atlas.scss');
+} else {
+	require('@clayui/css/src/scss/base.scss');
+}
+
+if (!devEnv || showSiteCss === 'true' || showSiteCss === null) {
+	require('./src/styles/main.scss');
+}
```
Comment (matuzalemsteles) [inline clayui.com/gatsby-browser.js:19]: Hey @bryceosterhaus, it looks like the pattern is showing only the base. Probably due to the above conditional that compared with the same value, I think `showAtlas` and `showSiteCss` is coming undefined.

**Comment by bryceosterhaus** [comment]
I can't seem to nail down why the base theme is importing and overriding. I think it might be a cache issue. I am going to close this PR and re-send to see if that does anything different

**Comment by bryceosterhaus** [comment]
new pr https://github.com/liferay/clay/pull/3362

**Pull request closed without merging**